<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>main_window</class>
 <widget class="QMainWindow" name="main_window">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>746</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Tumult</string>
  </property>
  <widget class="QWidget" name="central_widget">
   <layout class="QVBoxLayout" name="central_layout">
    <item>
     <widget class="QStackedWidget" name="central_stack">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="chat_page">
       <layout class="QVBoxLayout" name="chat_page_layout">
        <item>
         <layout class="QHBoxLayout" name="server_info_layout" stretch="1,0">
          <item>
           <widget class="QLabel" name="server_name_label">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="leave_button">
            <property name="text">
             <string>Leave</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QTextBrowser" name="chat_box"/>
        </item>
        <item>
         <layout class="QHBoxLayout" name="message_box_layout">
          <item>
           <widget class="QLineEdit" name="message_box_input">
            <property name="clearButtonEnabled">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="send_button">
            <property name="text">
             <string>Send</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="connect_page">
       <property name="autoFillBackground">
        <bool>false</bool>
       </property>
       <layout class="QVBoxLayout" name="connect_page_layout">
        <property name="spacing">
         <number>10</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <spacer name="connect_page_top_spacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item alignment="Qt::AlignHCenter">
         <widget class="QWidget" name="form_container" native="true">
          <property name="maximumSize">
           <size>
            <width>250</width>
            <height>16777215</height>
           </size>
          </property>
          <layout class="QVBoxLayout" name="form_container_layout">
           <property name="spacing">
            <number>10</number>
           </property>
           <property name="leftMargin">
            <number>7</number>
           </property>
           <property name="topMargin">
            <number>7</number>
           </property>
           <property name="rightMargin">
            <number>7</number>
           </property>
           <property name="bottomMargin">
            <number>7</number>
           </property>
           <item>
            <layout class="QFormLayout" name="connect_form_layout">
             <property name="sizeConstraint">
              <enum>QLayout::SetMinimumSize</enum>
             </property>
             <property name="fieldGrowthPolicy">
              <enum>QFormLayout::ExpandingFieldsGrow</enum>
             </property>
             <property name="labelAlignment">
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
             <property name="leftMargin">
              <number>0</number>
             </property>
             <property name="topMargin">
              <number>0</number>
             </property>
             <property name="rightMargin">
              <number>0</number>
             </property>
             <item row="0" column="0">
              <widget class="QLabel" name="nickname_label">
               <property name="text">
                <string>Nickname</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QLineEdit" name="nickname_input">
               <property name="maxLength">
                <number>10</number>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QLabel" name="server_address_label">
               <property name="text">
                <string>Server Address</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QLineEdit" name="server_address_input">
               <property name="maxLength">
                <number>15</number>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QLabel" name="server_port_label">
               <property name="text">
                <string>Server Port</string>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QLineEdit" name="server_port_input">
               <property name="maxLength">
                <number>5</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item alignment="Qt::AlignHCenter">
            <widget class="QPushButton" name="connect_button">
             <property name="maximumSize">
              <size>
               <width>125</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="text">
              <string>Connect</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <spacer name="connect_page_bottom_spacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
